src/SAWScript/Crucible/MIR/Builtins.hs

```diff
@@ -476,11 +533,34 @@ setupPrePointsTos ::
   [MirPointsTo] ->
   Crucible.SymGlobalState Sym ->
   IO (Crucible.SymGlobalState Sym)
-setupPrePointsTos _mspec _cc _env pts mem0 = foldM doPointsTo mem0 pts
+setupPrePointsTos mspec cc env pts mem0 = foldM doPointsTo mem0 pts
   where
-    doPointsTo :: Crucible.SymGlobalState Sym -> MirPointsTo -> IO (Crucible.SymGlobalState Sym)
-    doPointsTo _mem _pt =
-      panic "setupPrePointsTo" ["not yet implemented"]
+    tyenv = MS.csAllocations mspec
+    nameEnv = mspec ^. MS.csPreState . MS.csVarTypeNames
+
+    doPointsTo ::
+         Crucible.SymGlobalState Sym
+      -> MirPointsTo
+      -> IO (Crucible.SymGlobalState Sym)
+    doPointsTo globals (MirPointsTo _ allocIdx referents) =
+      mccWithBackend cc $ \bak -> do
+        referent <- firstPointsToReferent referents
+        MIRVal referentTy referentVal <-
+          resolveSetupVal cc env tyenv nameEnv referent
+        Some mp <- pure $ lookupAllocIndex env allocIdx
+        -- By the time we reach here, we have already checked (in mir_points_to)
+        -- that the type of the reference is compatible with the right-hand side
+        -- value, so the equality check below should never fail.
+        Refl <-
+          case W4.testEquality (mp^.mpType) (shapeType referentTy) of
+            Just r -> pure r
+            Nothing ->
+              panic "setupPrePointsTos"
+                    [ "Unexpected type mismatch between reference and referent"
+                    , "Reference type: " ++ show (mp^.mpType)
+                    , "Referent type:  " ++ show (shapeType referentTy)
+                    ]
+        Mir.writeMirRefIO bak globals Mir.mirIntrinsicTypes (mp^.mpRef) referentVal
```

Comment:
This is one of the places in the patch that involves the `crucible-mir` memory model. Here, we write the right-hand side values for each `mir_points_to` statement in a specification's preconditions.